delivery_postlogistics/postlogistics/web_service.py

```diff
@@ -0,0 +1,603 @@
+        """
+        return picking.carrier_id.postlogistics_output_format.code
+
+    def _get_image_resolution(self, picking):
+        """
+        Get Output Resolution Code define in carrier
+        """
+        return picking.carrier_id.postlogistics_resolution.code
+
+    def _get_license(self, picking):
+        """Get the license
+
+        Take it from carrier and if not defined get the first license.
+
+        :return: license number
+        """
+        franking_license = picking.carrier_id.postlogistics_license_id
+        return franking_license.number
+
+    def _prepare_attributes(
+        self, picking, pack=None, pack_num=None, pack_total=None, pack_weight=None
+    ):
+        package_type = (
+            pack
+            and pack.package_type_id
+            or picking.carrier_id.postlogistics_default_package_type_id
+        )
+        services = package_type._get_shipper_package_code_list()
+
+        if pack_weight:
+            total_weight = pack_weight
+        else:
+            total_weight = pack.shipping_weight if pack else picking.shipping_weight
+        total_weight *= 1000
+
+        if not services:
+            raise UserError(
+                picking.env._(
+                    "No PostLogistics packaging services found "
+                    "in package type {package_type_name}, for picking {picking_name}."
+                ).format(package_type_name=package_type.name, picking_name=picking.name)
+            )
+
+        # Activate phone notification ZAW3213
+        # if phone call notification is set on partner
+        if picking.partner_id.postlogistics_notification == "phone":
+            services.append("ZAW3213")
+
+        attributes = {
+            "weight": int(total_weight),
+        }
+
+        # Remove the services if the delivery fixed date is not set
+        if "ZAW3217" in services:
+            if picking.delivery_fixed_date:
+                attributes["deliveryDate"] = picking.delivery_fixed_date
+            else:
+                services.remove("ZAW3217")
+
+        # parcelNo / parcelTotal cannot be used if service ZAW3218 is not activated
+        if "ZAW3218" in services:
+            if pack_total > 1:
+                attributes.update(
+                    {"parcelTotal": pack_total - 1, "parcelNo": pack_num - 1}
+                )
+            else:
+                services.remove("ZAW3218")
+
+        if "ZAW3219" in services and picking.delivery_place:
+            attributes["deliveryPlace"] = picking.delivery_place
+        if picking.carrier_id.postlogistics_proclima_logo:
+            attributes["proClima"] = True
+        else:
+            attributes["proClima"] = False
+
+        attributes["przl"] = services
+
+        return attributes
+
+    def _get_itemid(self, picking, pack_no):
+        """Allowed characters are alphanumeric plus `+`, `-` and `_`
+        Last `+` separates picking name and package number (if any)
```

Comment:
I would give in argument the package instead of the package name.
Prefixing the package name with the picking name could be a parameter.
```suggestion
    def _get_itemid(self, picking, package):
        """ID assigned by customers at address label level.
        
        Allowed characters are alphanumeric plus `+`, `-` and `_`.
        Returns the picking name and package name (if any) separated by `+`. 
```